Library/src/Systems/Spaces/SpaceSystem.cpp

```diff
@@ -1945,4 +1963,41 @@ void SpaceSystem::DuplicateSpace(const String& SpaceId,
 	);
 }
 
+// expose a method to set a callback (DONE) and fire it when they get it (?)
+// add a callback for leaving scopes (DONE)
+
+// spacesystem to take a multiplayerconnection in the ctor (DONE)
+// in multiplayerconnection: add getspacesystem->seetconnection. 195. (DONE) and create a bind there (DONE)
+// create new id spacesystem leavescopes (DONE)
+// duplicate onrequesttodisconnct (DONE)
+// add a test in space system. enter a space, bind to the event, exit the space, add an await. (DONE??? fails. event not called)
+
+void SpaceSystem::BindOnRequestToLeaveScopes()
+{
+	Connection->On("OnRequestToLeaveScopes",
+				   [this](const signalr::value& Params)
+				   {
+					   auto Scopes				= Params.as_array()[0].as_array();
```

Comment:
This is the callback that I am trying (and failing) to call from the test

Reply:
Shooting in the dark here, but I wonder if because the hub method has 2 params on it (https://github.com/magnopus/Magnopus.Services/blob/071cfd4b7553d2314a09017456c607152f9f0a5a/Source/Magnopus.Service.Multiplayer.Contracts/HubClients/IMultiplayerHubClient.cs#L47) it might fail to find this?